### ISSUE TITLE
When dialog modal is opened, focus is not shifted to dialog modal when using VoiceOver on iOS

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#281]
# Bug Report

## Description
When dialog modal is opened, focus is not shifted to dialog modal when using VoiceOver on iOS

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-framework/#/tests/terra-dialog-modal/dialog-modal/default-dialog-modal
2. Open modal

## Expected Behavior
Focus should be shifted to open modal.

## Possible Solution
Programmatically set focus on dialog modal when opened via the use of refs on the modal dialog container.

## Environment

* Component Name and Version: terra-dialog-modal
* Browser Name and Version: Mobile Safari w/ VoiceOver on iOS 12

**Comment by emilyrohrbough** [comment]
We can verify the dialog modal has focus after opening the menu wit webdriver's [`hasFocus() command`](http://webdriver.io/api/state/hasFocus.html). We should add this to our tests.

**Comment by bjankord** [comment]
I've found the [notification dialog](https://engineering.cerner.com/terra-framework/#/components/terra-notification-dialog/notification-dialog/notification-dialog) seems to work much better on iOS using VoiceOver. Virtual focus is shifted to the notifiaction dialog and the entire content of the dialog is read to the user when its opened.

**Comment by JakeLaCombe** [comment]
It looks like role="alertdialog" is allowing for voiceover IOS to get that focus. I tested it just now from switching between the two roles, and it looks like that is adding extra support for focus.

**Comment by bjankord** [comment]
This has been resolved with https://github.com/cerner/terra-core/pull/2017
Tested on iOS with VO and focus is shifted and trapped within dialog modal.
